pkg/api/server/v1alpha2/results.go

```diff
@@ -67,6 +68,34 @@ func (s *Server) GetResult(ctx context.Context, req *pb.GetResultRequest) (*pb.R
 	return result.ToAPI(store), nil
 }
 
+// UpdateResult receives Result and FieldMask from client and uses them to update records in local Sqlite Server.
+func (s Server) UpdateResult(ctx context.Context, req *pb.UpdateResultRequest) (*pb.Result, error) {
+	// Retrieve result from database by name
+	tx := s.db.WithContext(ctx).Begin()
+	r, err := GetResultByName(tx, req.GetName())
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "failed to find a result: %v", err)
+	}
+	r.Annotations = req.GetResult().GetAnnotations()
+	r.UpdatedTime = timestamppb.New(clock.Now())
+
+	// Write result back to database.
+	dbResult, err := result.ToStorage(r)
+	if err != nil {
+		return nil, err
+	}
+	if err := errors.Wrap(tx.Save(&dbResult).Error); err != nil {
+		log.Printf("failed to save result into database: %v", err)
+		tx.Rollback()
+		return nil, err
+	}
+	if err := tx.Commit().Error; err != nil {
+		log.Printf("failed to commit transaction: %v", err)
+		return nil, fmt.Errorf("failed to commit transaction: %v", err)
```

Comment:
Using fmt.Errorf will remove the RPC status from the error, resulting in UNKNOWN being returned. You'll probably want to do something like:

```
status.Errorf(status.Code(err), "failed to commit transaction: %v", err)
```

Reply:
Got that, Done